### PR TITLE
Doctest extension

### PR DESCRIPTION
**Pull request by PerssonAlbin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,18 +31,31 @@ On macOS: `brew install cmake`
 
 ## Build
 
+To unit test build run:
+
+```shell
+mkdir build && cd build
+cmake -DCMAKE_BUILD_TYPE=Debug .
```
Comment (andreas-bauer) [inline README.md:38]: This command will not work since there is no CMake file in the `build` folder.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
+                        Tests in tests/*.cpp will still be enabled." ON)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
+include(ConfigSafeGuards)
+include(Doctest)
+
 set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -DDOCTEST_CONFIG_DISABLE")
-set(EXECUTABLE_OUTPUT_PATH bin/)
+set(EXECUTABLE_OUTPUT_PATH ../bin/)
```
Comment (andreas-bauer) [inline CMakeLists.txt:12]: I don't think that it is a good idea to have the executable in the `../bin` folder when every other build artifact is in the `build` folder, e.g. the `unit_tests`.

File: README.md [hunk 3]
```diff
@@ -31,18 +31,31 @@ On macOS: `brew install cmake`
 
 ## Build
 
+To unit test build run:
```
Comment (andreas-bauer) [inline README.md:34]: Debug build and unit test build could be combined in the README.

E.g.
```shell
mkdir build && cd build
cmake -DCMAKE_BUILD_TYPE=Debug .
make

# run unit tests
./unit_tests

# run debug build
./file_editor.out
```

**Comment by andreas-bauer** [comment]
Looks good if you want the seperation of concerns between app and unit tests.

File: CMakeLists.txt (hunk 2 above)
Comment (PerssonAlbin) [reply CMakeLists.txt]: All will now be built inside the build folder!

File: README.md (hunk 1 above)
Comment (PerssonAlbin) [reply README.md]: Rewrote the command in readme

File: README.md (hunk 3 above)
Comment (PerssonAlbin) [reply README.md]: Applied!

**Review by andreas-bauer — APPROVED** [review]
Looks good now 👍

**Pull request merged**